electrumsv/wallet_database/cache.py

```diff
@@ -128,29 +127,31 @@ def _add(self, inserts: List[TransactionRow],
         overwrite them.
         """
         date_added = self._store._get_current_timestamp()
-        for i, (tx_hash, metadata, bytedata, add_flags, description) in enumerate(inserts):
+        for i, (tx_hash, metadata, tx, add_flags, description) in enumerate(inserts):
             assert tx_hash not in self._cache, \
                 f"Tx {hash_to_hex_str(tx_hash)} found in cache unexpectedly"
             flags = self._adjust_metadata_flags(metadata, add_flags)
-            if bytedata is not None:
+            if tx is not None:
                 flags |= TxFlags.HasByteData
             assert ((add_flags & TxFlags.METADATA_FIELD_MASK) == 0 or flags == add_flags), \
                 f"{TxFlags.to_repr(flags)} != {TxFlags.to_repr(add_flags)}"
             self._validate_new_flags(tx_hash, flags)
             metadata = TxData(metadata.height, metadata.position, metadata.fee, date_added,
                 date_added)
             self._cache[tx_hash] = TransactionCacheEntry(metadata, flags)
-            if bytedata is not None:
-                self._bytedata_cache.set(tx_hash, bytedata)
-            inserts[i] = TransactionRow(tx_hash, metadata, bytedata, flags, description)
+            bytedata = None
+            if tx is not None:
+                self._txdata_cache.set(tx_hash, tx)
+                bytedata = tx.to_bytes()
+            inserts[i] = (tx_hash, metadata, bytedata, flags, description)
```

Comment:
Is there a reason this isn't still `= TransactionRow(tx_hash, metadata, bytedata, flags, description)`

Reply:
Because `bytedata` is not bytedata anymore. It's a `Transaction` and delaying the conversion avoids back and forth de/serialization steps

Reply:
It follows `tx.to_bytes()` so isn't bytedata is bytes again then?

Reply:
Actually this was the thing that was annoying me last night... I've changed it to `TransactionRow` and added # type: ignore... it's the only way. `inserts` is being modified in-place (with a type change) and with mypy's OCD tendencies... it won't have a bar of it. Casting isn't working for me either...